ArjunaJTA/jta/classes/com/arjuna/ats/jta/common/JTAEnvironmentBean.java

```diff
@@ -1225,4 +1236,127 @@ public void setNotifyCommitMarkableResourceRecoveryModuleOfCompleteBranches(
 			boolean notifyCommitMarkableResourceRecoveryModuleOfCompleteBranches) {
 		this.notifyCommitMarkableResourceRecoveryModuleOfCompleteBranches = notifyCommitMarkableResourceRecoveryModuleOfCompleteBranches;
 	}
+
+	/**
+     * Setting name of class which is taken to be used as transaction importer
+     * when a {@link javax.transaction.xa.Xid} needs to be imported as a transaction to currently running TM.<br>
+     * The classname which is set here has to implement {@link org.jboss.tm.SubordinateTransactionImporter} interface.
+     */
+    public void setSubordinateTransactionImporterClassName(String importerClassName) {
+        synchronized(this)
+        {
+            if(importerClassName == null)
+            {
+                this.subordinateTransactionImporter = null;
+            }
+            else if(!importerClassName.equals(this.subordinateTransactionImporterClassName))
+            {
+                this.subordinateTransactionImporter = null;
+            }
+            this.subordinateTransactionImporterClassName = importerClassName;
+        }
+    }
+
+    public String getSubordinateTransactionImporterClassName(){
+        return this.subordinateTransactionImporterClassName;
+    }
+
+    public void setSubordinateTransactionImporter(SubordinateTransactionImporter importer) {
+        synchronized(this) {
+            SubordinateTransactionImporter oldImporter = this.subordinateTransactionImporter;
+            this.subordinateTransactionImporter = importer;
+
+            if(importer == null)
+            {
+                this.subordinateTransactionImporterClassName = null;
+            }
+            else if(importer != oldImporter)
```

Comment:
What about my first question: "Did you intend using identity as opposed importer.equals(oldImporter)?"